lib/components/Radio/Radio.tsx

```diff
@@ -1,25 +1,59 @@
-import React, { forwardRef } from 'react';
+import React, { forwardRef, useContext } from 'react';
+import assert from 'assert';
 import {
   InlineField,
   InlineFieldProps,
 } from '../private/InlineField/InlineField';
+import { RadioGroupContext } from '../RadioGroup/RadioGroupContext';
+import dedent from 'dedent';
 
 export interface RadioProps
   extends Omit<
     InlineFieldProps,
     'message' | 'reserveMessageSpace' | 'required'
   > {}
 
-const NamedRadio = forwardRef<HTMLInputElement, RadioProps>((props, ref) => (
-  <InlineField
-    {...props}
-    type="radio"
-    message={null}
-    reserveMessageSpace={false}
-    required={undefined}
-    ref={ref}
-  />
-));
+const NamedRadio = forwardRef<HTMLInputElement, RadioProps>((props, ref) => {
+  const radioListContext = useContext(RadioGroupContext);
+
+  assert(
```

Comment:
This should be a `console.warn` (when not in prod) rather than an `assert`, otherwise it's a breaking change.

Reply:
It shouldnt be a breaking change, because existing `Radio` components cannot be inside a `RadioGroupContext` today.

Reply:
Ignore me, just realised this only throws when used inside `RadioGroup`.

Reply:
😄